test/unit/modules/accounts.js

```diff
@@ -0,0 +1,726 @@
+	secondPublicKey: null,
+	vote: '9820020609280331',
+	rate: '0',
+	delegates: null,
+	u_delegates: null,
+	multisignatures: null,
+	u_multisignatures: null,
+	multimin: 0,
+	u_multimin: 0,
+	multilifetime: 0,
+	u_multilifetime: 0,
+	blockId: '10352824351134264746',
+	nameexist: 0,
+	u_nameexist: 0,
+	producedblocks: 27,
+	missedblocks: 1,
+	fees: '231386135',
+	rewards: '0',
+	virgin: 1
+};
+
+describe('account', function () {
+
+	var account; 
+	var accountLogic;
+	var accountModuleDependencies;
+
+	before(function (done) {
+		async.auto({
+			rounds: function (cb) {
+				modulesLoader.initModule(Rounds, modulesLoader.scope,cb);
+			},
+			accountLogic: function (cb) {
+				modulesLoader.initLogicWithDb(AccountLogic, cb);
+			},
+			blockModule: ['accountLogic', function (result, cb) {
+				modulesLoader.initModuleWithDb(BlocksModule, cb, {
+					logic : { /* dependencies not included */},
+				});
+			}],
+			transactionLogic: ['accountLogic', function (result, cb) {
+				modulesLoader.initLogicWithDb(TransactionLogic, cb, {
+					account: result.accountLogic
+				});
+			}],
+			delegateModule: ['transactionLogic', function (result, cb) {
+				modulesLoader.initModuleWithDb(DelegateModule, cb, {
+					logic: {
+						transaction: result.transactionLogic
+					}
+				});
+			}],
+			transactionModule: ['transactionLogic', function (result, cb) {
+				modulesLoader.initModuleWithDb(TransactionModule, cb, {
+					transaction: result.transactionLogic
+				});
+			}]
+		}, function (err, result) {
+			modulesLoader.initModuleWithDb(AccountModule, function (err, __accountModule) {
+				expect(err).to.not.exist;
+
+				account = __accountModule;
+				accountLogic = result.accountLogic;
+
+				// for correctly initializing setting blocks module
+				result.blockModule.lastBlock.set({height: 10});
+
+				result.delegateModule.onBind({
+					accounts: __accountModule,
+					transactions: result.transactionModule,
+					blocks: result.blockModule
+				});
+
+				result.transactionModule.onBind({
+					accounts: __accountModule,
+					transactions: result.transactionModule,
+					//loader: 
+				});
+
+				account.onBind({
+					delegates: result.delegateModule,
+					accounts: account,
+					transactions: result.transactionModule
+				});
+
+				accountModuleDependencies = result; 
+				done();
+			}, {
+				logic: {
+					account: result.accountLogic,
+					transaction: result.transactionLogic
+				}
+			});
+		});
+	});
+
+	describe('Accounts', function () {
+		it('should throw with no params', function () {
+			expect(function () {
+				new AccountModule();
+			}).to.throw();
+		});
+	});
+
+	describe('generateAddressByPublicKey', function () {
+		it('should generate correct address for the publicKey provided', function () {
+			expect(account.generateAddressByPublicKey(validAccount.publicKey)).to.equal(validAccount.address);
+		});
+
+		it.skip('should throw error for invalid publicKey', function () {
+			var invalidPublicKey = 'invalidPublicKey';
```

Comment:
you could add a test for proper publicKey length in `Accounts.prototype.generateAddressByPublicKey` function